Feature/align config RCS-111

kuenishi opened the pull request:
c.f. [Configuration Mapping Table between 1.5 and 2.0](https://github.com/basho/riak_cs/wiki/Configuration-Mapping-Table-between-1.5-and-2.0-%5BRFC%5D)

kuenishi commented:
@shino @angrycub it's ready for review; could you take a look at this?

kuenishi commented on rel/files/riak_cs.schema:
```diff
@@ -154,7 +156,6 @@
 %% This leeway time is set to give the delete
 %% indication time to propogate to all replicas.
-%% 86400 is 24-hours.
 {mapping, "gc.leeway_seconds", "riak_cs.leeway_seconds", [
```
`leeway_period` because `leeway_seconds = 24h` is somewhat stupid.

kuenishi replied:
Changed to leeway_period

shino commented on rel/files/riak_cs.schema:
```diff
@@ -181,7 +180,14 @@
+]}.
+
+{mapping, "gc.max_workers", "riak_cs.gc_max_workers", [
+  {default, 5},
```
How about changing default value? Five workers can make rather large load. I want to recommend 2 to begin with.

kuenishi replied:
Changed to 2

shino commented on rel/files/riak_cs.schema:
```diff
@@ -189,7 +195,7 @@
 %% access_archive_period (1 == once per period; 2 == twice per period,
 %% etc.)
-{mapping, "access.stats.log_flush_factor", "riak_cs.access_log_flush_factor", [
+{mapping, "access.stats.log.flush_factor", "riak_cs.access_log_flush_factor", [
```
There are two kinds of statistics in riak cs, access and storage. If they are grouped, `stats.access.*` is better name.
If not,  `access_stats` is nice for prefix.

shino commented on rel/files/riak_cs.schema:
```diff
@@ -265,7 +270,7 @@
 %% @doc custom server name at http response header "Server: Riak CS"
 {mapping, "server.name", "webmachine.server_name", [
-  {default, "Riak CS"},
+  {default, "Riak CS" },
```
The space added?

kuenishi replied:
I tried to introduce version number here but didn't work well.

shino replied:
Why not remove the space?

shino commented:
cuttlefish 2.0.1 had been tagged. Update?

shino commented:
Comments which are not for diff in this PR but cuttlefish'ization related.
- `riak_cs.conf` or `riak-cs.conf`? (underscore or hyphen)
  I want to +1 for hyphen because the command `riak-cs` uses hyphen.
- `dev_vars.confg` needs `devrel` flag:
  cf. https://github.com/basho/riak/blob/develop/rel/vars/dev_vars.config.src#L4
- Why is `cs_port` restricted to > 1024?
  If there is no strong reason, operators can use well-known ports at their own risk.
- I want `admin.ip` and `admin.port` is commented out, rather than hidden.
- Is it better to add `admin.` prefix for `anonymous_user_creation`?
  Operators will know this config item is about administration.
- How about introduce prefix `api.`? (or `s3_api.`?)
  e.g. `api.root_host`, `api.trust_x_forwared_for`, `api.region`, ...
- It's better to change `pool.*.overflow`'s hidden (because we don't test them)
- `request_limit` is defined with the prefix `storage.`.
  But `riak_cs.usage_request_limit` affects both storage and access.
- `storage.stats.request_limit` is defined by `{duration, h}`.
  On the other hand, `riak_cs.usage_request_limit` is used as count or just a number
  (not a duration).
  These two are coincide (accidentally) only if `storage.stats.archive_period`
  and `access.stats.archive_period` are `1h`.
- If we align to riak, `dtrace_support` will be `dtrace` as in `riak_core.schema`:
  cf. https://github.com/basho/riak_core/blob/develop/priv/riak_core.schema#L135-L142
- As well as `riak.schema`, it's good to enable `sfwi` and disable `scl`:
  https://github.com/basho/riak/blob/2.0/rel/files/riak.schema#L208-L219

kuenishi commented:
> `riak_cs.conf` or `riak-cs.conf`? (underscore or hyphen)

Changed

> `dev_vars.confg` needs `devrel` flag:

Added

> Why is `cs_port` restricted to > 1024?

Changed to > 0 

> I want `admin.ip` and `admin.port` is commented out, rather than hidden.

Commented out

> Is it better to add `admin.` prefix for `anonymous_user_creation`?

I don't think so. Too much categories..

> How about introduce prefix `api.`? (or `s3_api.`?)

Same above

> It's better to change `pool.*.overflow`'s hidden (because we don't test them)

Hidden.

> `request_limit` is defined with the prefix `storage.`.

Removed prefix and changed to `stats.usage_request_limit`

> `storage.stats.request_limit` is defined by `{duration, h}`.
>  On the other hand, `riak_cs.usage_request_limit` is used as count or just a number  (not a duration).
>  These two are coincide (accidentally) only if `storage.stats.archive_period` and `access.stats.archive_period` are `1h`.

Description updated

> If we align to riak, `dtrace_support` will be `dtrace` as in `riak_core.schema`:

Changed

> As well as `riak.schema`, it's good to enable `sfwi` and disable `scl`:

Added

kuenishi commented:
Cuttlefish is also updated to 2.0.1.

kuenishi commented:
TODO: update configuration mapping and open as another pull request.

shino commented on rel/files/riak_cs.schema:
```diff
@@ -150,19 +152,16 @@
+%% @doc The time to retain the block for an object after it has been
+%% deleted.  This leeway time is set to give the delete indication
+%% time to propogate to all replicas.
+{mapping, "gc.leeway_seconds", "riak_cs.leeway_period", [
```
`gc.leeway_seconds` ?

kuenishi replied:
heh

shino commented:
> riak_cs.conf or riak-cs.conf? (underscore or hyphen)

`cuttlefish_conf` in `vars.config` should also be changed.

kuenishi commented:
Updated. Can we postpone work on +sfwi stuff? Looks like they're not working correctly and maybe involve cuttlefish.

shino commented:
> Can we postpone work on +sfwi stuff? Looks like they're not working correctly and maybe involve cuttlefish.

:accept:

shino commented:
All riak_test cases passed.

kuenishi commented:
@borshop merge
